submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/StatsMongoDataService.kt

```diff
@@ -5,57 +5,77 @@ import ac.uk.ebi.biostd.persistence.common.model.SubmissionStat
 import ac.uk.ebi.biostd.persistence.common.model.SubmissionStatType
 import ac.uk.ebi.biostd.persistence.common.request.PaginationFilter
 import ac.uk.ebi.biostd.persistence.common.service.StatsDataService
-import ac.uk.ebi.biostd.persistence.doc.db.data.SubmissionStatsDataRepository
-import ac.uk.ebi.biostd.persistence.doc.model.DocSubmission
+import ac.uk.ebi.biostd.persistence.doc.db.repositories.SubmissionMongoRepository
+import ac.uk.ebi.biostd.persistence.doc.db.repositories.SubmissionStatsRepository
+import ac.uk.ebi.biostd.persistence.doc.model.DocStat
+import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionStats
 import ac.uk.ebi.biostd.persistence.doc.model.SingleSubmissionStat
+import org.bson.types.ObjectId
 import org.springframework.data.domain.PageRequest
 
 class StatsMongoDataService(
-    private val statsDataRepository: SubmissionStatsDataRepository
+    private val statsRepository: SubmissionStatsRepository,
+    private val submissionsRepository: SubmissionMongoRepository,
 ) : StatsDataService {
     override fun findByType(submissionStatType: SubmissionStatType, filter: PaginationFilter): List<SubmissionStat> =
-        statsDataRepository
+        statsRepository
             .findAllByStatType(submissionStatType, PageRequest.of(filter.pageNumber, filter.limit))
             .content
             .map { toSubmissionStat(it, submissionStatType) }
 
     override fun findByAccNoAndType(accNo: String, submissionStatType: SubmissionStatType): SubmissionStat =
-        statsDataRepository
+        statsRepository
             .findByAccNoAndStatType(accNo, submissionStatType)
             ?.let { toSubmissionStat(it, submissionStatType) }
             ?: throw StatNotFoundException(accNo, submissionStatType)
 
-    // TODO support saving stats for expired versions Pivotal ID # 176802128
-    override fun saveAll(stats: List<SubmissionStat>): List<SubmissionStat> =
-        stats.filterValid()
-            .mapValues { it.value.last() }
-            .map { updateSubmissionStat(it.key!!, it.value) }
+    override fun save(stat: SubmissionStat): SubmissionStat {
+        updateOrRegister(stat)
+        return stat
+    }
+
+    override fun saveAll(stats: List<SubmissionStat>): List<SubmissionStat> {
+        val valid = stats.filterValid().map { it.value.last() }
+        valid.forEach { updateOrRegister(it) }
```

Comment:
we can use db power to save then at single time instead of hitting database many times.

Reply:
As far as I understand, there's no way to perform upserts on array elements in mongo